src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscalesConfiguration.hooks.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect } from 'react';
+import { useFormContext, useFieldArray } from 'react-hook-form';
+
+import { useCurrentActivity } from 'modules/Builder/pages/BuilderApplet/BuilderApplet.hooks';
+import { ItemFormValues } from 'modules/Builder/pages';
+import { ActivitySettingsSubscale } from 'shared/state';
+
+import { ageItem, genderItem } from './SubscalesConfiguration.const';
+
+export const useSubscalesSystemItemsSetup = () => {
+  const { fieldName: activityFieldName } = useCurrentActivity();
+  const { control, watch } = useFormContext();
+  const itemsFieldName = `${activityFieldName}.items`;
+  const items: ItemFormValues[] = watch(itemsFieldName);
+  const { append: appendItem, remove: removeItem } = useFieldArray({
+    control,
+    name: itemsFieldName,
+  });
+  const subscalesField = `${activityFieldName}.subscales`;
+  const subscales: ActivitySettingsSubscale[] = watch(subscalesField) ?? [];
+
+  useEffect(() => {
+    const hasSubscaleLookupTable = subscales.some((subscale) => !!subscale.subscaleTableData);
+    const subscaleSystemItems = items.reduce((acc, item, index) => {
+      if (item.isSubscaleSystemItem) return acc.concat({ ...item, index });
```

Comment:
Maybe this return is stopping the cycle or it has to be so?

Reply:
actually, this `reducer` filters the system items and add the base array `index` to each item